src/Microsoft.AspNet.Mvc.WebApiCompatShim/Formatters/HttpResponseMessageOutputFormatter.cs

```diff
@@ -59,6 +68,12 @@ public bool CanWriteResult(OutputFormatterContext context, MediaTypeHeaderValue
                 if (responseMessage.Content != null)
                 {
                     var contentHeaders = responseMessage.Content.Headers;
+                    
+                    // Copy the response content headers only after ensuring they are complete.
+                    // We ask for Content-Length first because HttpContent lazily computes this
+                    // and only afterwards writes the value into the content headers.
+                    var unused = contentHeaders.ContentLength;
```

Comment:
In Web API, this piece of code was put in try-catch, but I did not add it here as our story for exception handling and logging is different in K...